src/SmartComponents/Systems/System.test.js

```diff
@@ -1,171 +1,168 @@
+import configureStore from 'redux-mock-store';
+import { exportSystemsCSV, exportSystemsJSON } from '../../Utilities/api';
+import { systemRows } from '../../Utilities/RawDataForTesting';
+import { initMocks } from '../../Utilities/unitTestingUtilities.js';
+import Systems from './Systems';
+import toJson from 'enzyme-to-json';
+
+initMocks();
+
+jest.mock('../../store', () => ({
+    ...jest.requireActual('../../store'),
+    getStore: jest.fn(() => ({
+        getState: () => ({ SystemsListStore: { rows: [] } })
+    })),
+    register: jest.fn()
+}));
+
+jest.mock('react-redux', () => ({
+    ...jest.requireActual('react-redux'),
+    useSelector: jest.fn()
+}));
+
+jest.mock('@redhat-cloud-services/frontend-components-utilities/helpers', () => ({
+    ...jest.requireActual('@redhat-cloud-services/frontend-components-utilities/helpers'),
+    downloadFile: jest.fn()
+}));
+
+jest.mock('@redhat-cloud-services/frontend-components/Inventory', () => ({
+    ...jest.requireActual('@redhat-cloud-services/frontend-components/Inventory'),
+    InventoryTable: jest.fn(() => <div className='testInventroyComponentChild'><div>This is child</div></div>)
+}));
+
+jest.mock('../../Utilities/api', () => ({
+    ...jest.requireActual('../../Utilities/api'),
+    exportSystemsCSV: jest.fn(() => Promise.resolve({ success: true }).catch((err) => console.log(err))),
+    exportSystemsJSON: jest.fn(() => Promise.resolve({ success: true }).catch((err) => console.log(err))),
+    fetchSystems: jest.fn(() => Promise.resolve({ success: true }).catch((err) => console.log(err))),
+    fetchViewAdvisoriesSystems: jest.fn(() => Promise.resolve({ success: true }).catch((err) => console.log(err))),
+    fetchApplicableSystemAdvisoriesApi: jest.fn(() => Promise.resolve({
+        data: [{
+            attributes: {
+                advisory_type: 2,
+                description: 'The tzdata penhancements.',
+                public_date: '2020-10-19T15:02:38Z',
+                synopsis: 'tzdata enhancement update'
+            },
+            id: 'RHBA-2020:4282',
+            type: 'advisory'
+        }]
+    }).catch((err) => console.log(err)))
+}));
+
+const mockState = {
+    entities: {
+        rows: systemRows,
+        metadata: {
+            limit: 25,
+            offset: 0,
+            total_items: 10
+        },
+        expandedRows: {},
+        selectedRows: { 'RHSA-2020:2774': true },
+        error: {},
+        status: 'resolved'
+    },
+    GlobalFilterStore: {},
+    SystemsStore: {
+        queryParams: {}
+    }
+};
+const initStore = (state) => {
+    const customMiddleWare = () => next => action => {
+        useSelector.mockImplementation(callback => {
+            return callback(state);
+        });
+        next(action);
+    };
+
+    const mockStore = configureStore([customMiddleWare]);
+    return mockStore(state);
+};
+
+let wrapper;
+let store = initStore(mockState);
+
+beforeEach(() => {
```

Comment:
These could go `beforeEach` & `afterEach` can go within the describe block, or?

Reply:
Yup